src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs

```diff
@@ -285,7 +302,17 @@ private bool ShouldAnalyze(IOperation operationBlock, ISymbol owningSymbol)
                         return false;
                     }
 
-                    //  6. Otherwise, we execute analysis by walking the reaching symbol write chain to attempt to
+                    //  6. Bail out if we have a delegate wrapped in a tuple.
+                    //     This indicates the delegate targets (such as lambda/local functions) have been wrapped
+                    //     within a tuple and passed around and invoked, and these invocations can read values written
+                    //     to any local/parameter in the current method. We cannot reliably flag any write to a 
+                    //     local/parameter as unused for such cases.
+                    if (_delegateWrappedInTuple)
```

Comment:
We have a few different flags here related to having delegates that might escape, is that possible simplify it by combining them into a single flag?

Reply:
I had that initially implemented it with a single flag, but split it out into separate flags for clarity and better documentation.